EIPS/eip-1959.md

```diff
@@ -0,0 +1,70 @@
+---
+eip: 1959
+title: Valid ChainID opcode
+author: Ronan Sandford (@wighawag)
+category: Core
+type: Standards Track
+discussions-to: https://ethereum-magicians.org/t/eip-1959-valid-chainid-opcode/3170
+status: Draft
+created: 2019-04-20
+---
+
+## Abstract
+This EIP adds an opcode that returns whether the specific number passed in has been a valid chainID (EIP-155 unique identifier) in the history of the chain (including the current chainID).
+
+## Motivation
+[EIP-155](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md) proposes to use the chain ID to prevent replay attacks between different chains. It would be a great benefit to have the same possibility inside smart contracts when handling signatures, especially for Layer 2 signature schemes using [EIP-712](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md).
+
+[EIP-1344](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1344.md) is attempting to solve this by giving smart contract access to the tip of the chainID history. This is insuficient as such value is changing. Hence why EIP-1344 describes a contract based solution to work arround the problem. It would be better to solve it in a simpler, cheaper and safer manner, removing the potential risk of misuse present in EIP-1344.
+
+## Specification
+Adds a new opcode ```VALID_CHAINID``` at 0x46, which uses 1 stack argument : a uint256 value that represent the chainID to test. It will push ```true``` onto the stack if the uint256 value is part of the history of chainIDs of that chain, ```false``` otherwise. 
```

Comment:
This uses the same opcode as 1344. Perhaps it is worth highlighting somewhere that this is competing with 1344 and only one should be accepted?

Reply:
It doesn't necessarily have to compete in my opinion, but the author disagrees. I think regardless it should have a separate opcode

Reply:
As this proposal does not give access to the blockNumber at which chainId became history, in any form, having both #1344 and #1959 might be beneficial for some use case. I am not sure but if it was the case, it might be a clear indication that what we want is #1965.

As for opcode number, could this not be over specification:

if I say 0x46 and both #1344 and #1959 are included, this will cause trouble but if I say 0x47 (or whatever is available next) and only #1959 is included, this will cause an inelegant gap.

Reply:
I feel like in general, specifying what opcode number is used might be above and beyond what an EIP should capture, since really it's up to the fork that it's implemented in to set the opcode to whatever final number it gets set to. Sort of an implementation detail in my opinion.